### PR TITLE
feat(datepicker): Add invalid class for datepicker

### PR DESCRIPTION
**Pull request by mhchen**
Add a red border to the bottom of the datepicker input if the input
string is not a valid and complete date.

### CODE REVIEW

**Comment by mhchen** [comment]
@jelbourn Please review at your convenience.

File: src/components/datepicker/datePicker.js [hunk 1]
```diff
@@ -214,7 +216,10 @@
 
       if (self.dateUtil.isValidDate(parsedDate) && self.dateLocale.isDateComplete(inputString)) {
         self.ngModelCtrl.$setViewValue(parsedDate);
+        self.inputContainer.classList.remove(INVALID_CLASS);
         self.$scope.$apply();
+      } else if (inputString) {
+        self.inputContainer.classList.add(INVALID_CLASS);
```
Comment (jelbourn) [inline src/components/datepicker/datePicker.js]: If you go from the invalid state to an empty string, in invalid state doesn't go away; the invalid class should always go away if the input is empty.

**Comment by mhchen** [comment]
@jelbourn Sorry about that, rebased, please re-review

**Comment by jelbourn** [comment]
LGTM, merging...

**Comment by jelbourn** [comment]
Closed via 52fb2d1a5c086b8f37a75dac7c16e6935bcef14e

**Pull request closed without merging**